File: SeriesGuide/src/main/java/com/battlelancer/seriesguide/ui/PersonActivity.java
package com.battlelancer.seriesguide.ui;

import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.ActionBar;
import android.view.MenuItem;
import android.view.Window;
import com.battlelancer.seriesguide.R;

/**
 * Hosts a {@link com.battlelancer.seriesguide.ui.PersonFragment}, only used on handset devices. On
 * tablet-size devices a two-pane layout inside {@link com.battlelancer.seriesguide.ui.PeopleActivity}
 * is used.
 */
public class PersonActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        supportRequestWindowFeature(Window.FEATURE_ACTION_BAR_OVERLAY);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_person);
        setupActionBar();

        if (savedInstanceState == null) {
            PersonFragment f = PersonFragment.newInstance(
                    getIntent().getIntExtra(PersonFragment.InitBundle.PERSON_TMDB_ID, 0));
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.containerPerson, f)
                    .commit();
        }
    }

    @Override
    protected void setupActionBar() {
        super.setupActionBar();
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setBackgroundDrawable(
                    ContextCompat.getDrawable(this, R.drawable.background_actionbar_gradient));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowTitleEnabled(false);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == android.R.id.home) {
            onBackPressed();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/people/PersonActivity.kt
// SPDX-License-Identifier: Apache-2.0
// Copyright 2014-2024 Uwe Trottmann

package com.battlelancer.seriesguide.people

import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updatePaddingRelative
import com.battlelancer.seriesguide.R
import com.battlelancer.seriesguide.ui.BaseActivity
import com.battlelancer.seriesguide.util.ThemeUtils
import com.battlelancer.seriesguide.util.commitReorderingAllowed

/**
 * Hosts a [PersonFragment], only used on handset devices. On
 * tablet-size devices a two-pane layout inside [PeopleActivity]
 * is used.
 */
class PersonActivity : BaseActivity() {

    override fun configureEdgeToEdge() {
        // Always using a toolbar with dark to transparent gradient background,
        // so always use a status bar for dark backgrounds.
        ThemeUtils.configureEdgeToEdge(window, forceDarkStatusBars = true)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_person)
        ThemeUtils.configureForEdgeToEdge(findViewById(R.id.rootLayoutPerson))
        setupActionBar()

        // fitsSystemWindows="true" also adds padding for the navigation bar,
        // so manually readjust padding for just the status bar.
        val toolbar = findViewById<Toolbar>(R.id.sgToolbar)
        val toolbarPaddingTop = toolbar.paddingTop
        ViewCompat.setOnApplyWindowInsetsListener(toolbar) { v, insets ->
            val statusBarInsets = insets.getInsets(WindowInsetsCompat.Type.statusBars())
            v.updatePaddingRelative(top = toolbarPaddingTop + statusBarInsets.top)
            insets
        }

        if (savedInstanceState == null) {
            val f = PersonFragment.newInstance(
                intent.getIntExtra(PersonFragment.ARG_PERSON_TMDB_ID, 0)
            )
            supportFragmentManager.commitReorderingAllowed {
                add(R.id.containerPerson, f)
            }
        }
    }

    override fun setupActionBar() {
        super.setupActionBar()
        supportActionBar?.also {
            it.setHomeAsUpIndicator(R.drawable.ic_clear_24dp)
            it.setDisplayHomeAsUpEnabled(true)
            it.setDisplayShowTitleEnabled(false)
        }
    }

}
